### PR TITLE
refactor: dedupe `CallRequest`/`TransactionRequest`

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

`CallRequest` and `TransactionRequest` are basically the same type, with very few differences:

- `TransactionRequest` was missing an optional chain ID
- `TransactionRequest` was missing `max_fee_per_blob_gas`
- `CallRequest` and `TransactionRequest` had different blob fields

Since they also serve the same purpose, this PR dedupes them, to make sure there is only one tx request type, which is important for the network abstraction.

## Solution

Dedupe them, porting over fields missing from either.

I removed the blob transaction sidecar, but only because I didn't really know if it is actually a part of the spec. As far as I could tell, only `blob_versioned_hashes` and the gas field is. I asked @mattsse for direction in case he knew, but ultimately we didn't land on anything.

This is a breaking change for obvious reasons, and it not simply a "import new type" migration path, as I also ported over `CallInput` for `TransactionRequest`

## PR Checklist

- [x] Added Tests
- [x] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

**Comment by onbjerg** [comment]
@DaniPopes where do I note breaking changes for the checklist?

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/request.rs:40]: we still need this if we want to use this to convert into a full eip4844 transaction.

it's not needed for anything rpc besides eth_sendTransaction, but if we wan to use this type as the builder for constructing sendable transactions we need this

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/request.rs:46]: this should use order as CallRequest before, should come after nonce

**Review by mattsse — CHANGES_REQUESTED** [review]
I'm for unifying them, not worth having duplicate versions of the same thing. especially if we want to use this type as the foundation for building transactions.

before we merge this here, I'd like companion prs on foundry and reth so we can merge them in one go.

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (onbjerg) [reply crates/rpc-types/src/eth/transaction/request.rs:40]: thanks, that's what i was asking about in alloy maintainers, will add it back.

**Review by Evalir — APPROVED** [review]
This is fine w/ me. I'll take care of the foundry companion PR

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/request.rs:50]: this should also be #[serde(skip_serializing_if = "Option::is_none")]

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/request.rs:58]: why can't we derive this?

**Review by mattsse — CHANGES_REQUESTED** [review]
last nit

File: crates/rpc-types/src/eth/transaction/request.rs
Comment (onbjerg) [reply crates/rpc-types/src/eth/transaction/request.rs:58]: `OtherFields` contains `serde_json::Value` which is not `Hash`

**Comment by Evalir** [comment]
gm @onbjerg can this be merged?

**Pull request merged**
